mllib/src/main/scala/org/apache/spark/ml/evaluation/MulticlassClassificationEvaluator.scala

```diff
@@ -80,11 +97,22 @@ class MulticlassClassificationEvaluator @Since("1.5.0") (@Since("1.5.0") overrid
         case Row(prediction: Double, label: Double) => (prediction, label)
       }
     val metrics = new MulticlassMetrics(predictionAndLabels)
-    val metric = $(metricName) match {
-      case "f1" => metrics.weightedFMeasure
-      case "weightedPrecision" => metrics.weightedPrecision
-      case "weightedRecall" => metrics.weightedRecall
-      case "accuracy" => metrics.accuracy
+    val labelVal = getLabelValue
```

Comment:
The errors thrown here will end up being quite cryptic - `MatchError` will be thrown. It would be better to do some checks on consistency of the metric and label value, and throw with a more specific error message.